### PR TITLE
[bitnami/keycloak] Allow to use the same JKS keystore

### PR DESCRIPTION
**Pull request by yesteph**
**Description of the change**

If TLS termination is enables at the POD level, allow to use the same JKS keystore file for all the replicas.
Old mode with one shared truststore plus one keystore for each replica is kept and default.

**Benefits**

Simplify the creation of the Secret storing the JKS files.

Instead of a Secret with a key `keycloak.truststore.jks` plus a set of `keycloak-X.keystore.jks` where X is the ID of each Keycloak replica, you can now provide a Secret with two files: `truststore.jks`and `keystore.jks`.

**Possible drawbacks**

It may be a security concern to get a proper keystore per Keycloak replica if the keycloak pods are directly exposed to the end user traffic.
But this is generally not the case as we generally use an ingress or a level-7 load balancer.

**Applicable issues**

N/A

**Additional information**

JKS files key in the Secret (jksSecret) are renamed to match the keys generated by cert-manager when `keystores` are indicated in the Certificate resource.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/keycloak/templates/statefulset.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ spec:
         {{- if .Values.initContainers }}
         {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
         {{- end }}
-        {{- if .Values.auth.tls.enabled }}
+        {{- if and .Values.auth.tls.enabled (not .Values.auth.tls.sharedJKS) }}
```
Comment (javsalgar) [inline bitnami/keycloak/templates/statefulset.yaml:70]: Question, don't we want to change the permissions for the shared JKS certificates as well? Wouldn't this cause issues?

**Review by javsalgar — COMMENTED** [review]
Thank you so much for the PR! One question

File: bitnami/keycloak/templates/statefulset.yaml (hunk 1 above)
Comment (yesteph) [reply bitnami/keycloak/templates/statefulset.yaml:70]: Hello @javsalgar 

The files are just read. No special permissions required for that.

File: bitnami/keycloak/templates/configmap-env-vars.yaml [hunk 2]
```diff
@@ -32,6 +32,6 @@ data:
   KEYCLOAK_ENABLE_TLS: {{ ternary "true" "false" .Values.auth.tls.enabled | quote }}
   {{- if .Values.auth.tls.enabled }}
   KEYCLOAK_HTTPS_PORT: {{ .Values.containerPorts.https | quote }}
-  KEYCLOAK_TLS_KEYSTORE_FILE: "/opt/bitnami/keycloak/certs/keycloak.keystore.jks"
-  KEYCLOAK_TLS_TRUSTSTORE_FILE: "/opt/bitnami/keycloak/certs/keycloak.truststore.jks"
+  KEYCLOAK_TLS_KEYSTORE_FILE: {{ ternary "/certs/keystore.jks" "/opt/bitnami/keycloak/certs/keycloak.keystore.jks" .Values.auth.tls.sharedJKS | quote }}
```
Comment (javsalgar) [inline bitnami/keycloak/templates/configmap-env-vars.yaml:35]: I have my concerns about the paths. It seems to me that, for consistency, the other mount folder should also be `/opt/bitnami/keycloak`. Maybe `/opt/bitnami/keycloak/shared-certs`?

File: bitnami/keycloak/templates/statefulset.yaml (hunk 1 above)
Comment (javsalgar) [reply bitnami/keycloak/templates/statefulset.yaml:70]: Ok

**Pull request closed without merging**